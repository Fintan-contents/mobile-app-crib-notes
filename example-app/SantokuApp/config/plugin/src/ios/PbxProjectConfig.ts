import {ExportedConfigWithProps, XcodeProject} from 'expo/config-plugins';

import {PbxProject, PbxProjectXCBuildConfigurationId} from './types/pbxproj';

const getPbxProject = (config: ExportedConfigWithProps<XcodeProject>): PbxProject => {
  return config.modResults as PbxProject;
};

const getBuildConfigurations = (pbxProject: PbxProject): PbxProjectXCBuildConfigurationId[] => {
  const objects = pbxProject.hash.project.objects;
  const xcBuildConfigurations = objects.XCBuildConfiguration;
  const xcBuildConfigurationIds = Object.keys(xcBuildConfigurations);

  return xcBuildConfigurationIds
    .filter(id => !id.endsWith('_comment'))
    .map(id => ({
      [id]: {
        ...xcBuildConfigurations[id],
      },
    }));
};

const mergeBuildConfigurations = (pbxProject: PbxProject, buildConfigurations: PbxProjectXCBuildConfigurationId) => {
  // pbxProjectのprototypeを消さないように、Object.assignでマージしています。
  return Object.assign(pbxProject, {
    ...pbxProject,
    hash: {
      ...pbxProject.hash,
      project: {
        ...pbxProject.hash.project,
        objects: {
          ...pbxProject.hash.project.objects,
          XCBuildConfiguration: {
            ...pbxProject.hash.project.objects.XCBuildConfiguration,
            ...buildConfigurations,
          },
        },
      },
    },
  });
};

export const PbxProjectConfig = {
  getPbxProject,
  getBuildConfigurations,
  mergeBuildConfigurations,
};
