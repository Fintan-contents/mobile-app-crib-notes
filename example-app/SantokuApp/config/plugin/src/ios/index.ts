export * from './withIosRemoveCFBundleUrlTypes';
export * from './withIosEnabledATS';
export * from './withIosDisabledATS';
export * from './withIosSetCredentials';
export * from './withIosAddPersonalAccountConfig';
export * from './withIosOverrideStoryboard';
