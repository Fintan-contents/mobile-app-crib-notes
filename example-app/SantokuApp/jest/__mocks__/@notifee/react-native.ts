import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

// テストケースごとにモックは初期化しておく。
beforeEach(() =>
  Object.values(mock).forEach(fn => {
    if (jest.isMockFunction(fn)) {
      fn.mockClear();
    }
  }),
);

export default mock;
