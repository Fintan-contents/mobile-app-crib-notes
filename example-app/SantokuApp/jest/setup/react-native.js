// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
