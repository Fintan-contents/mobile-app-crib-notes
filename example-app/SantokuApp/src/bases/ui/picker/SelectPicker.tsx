import React from 'react';
import {Pressable, StyleSheet, TextInput, TextInputProps as RNETextInputProps, View, ViewProps} from 'react-native';

import {DefaultPickerAccessory, DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdrop, PickerBackdropProps} from './PickerBackdrop';
import {PickerContainer, PickerContainerProps} from './PickerContainer';
import {SelectPickerItems, SelectPickerItemsProps} from './SelectPickerItems';
import {useSelectPicker} from './useSelectPicker';

export type Item<T> = {
  label: string;
  value: T;
  inputLabel?: string;
  key?: React.Key;
  color?: string;
  fontFamily?: string;
};

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;
export type SelectPickerProps<ItemT> = {
  /**
   * SelectPickerに表示するアイテム
   */
  items: Item<ItemT>[];
  /**
   * 選択されたアイテムのKey
   */
  selectedItemKey?: React.Key | ItemT;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (selectedItem?: Item<ItemT>) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   */
  onDismiss?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDelete?: (selectedItem?: Item<ItemT>) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onCancel?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDone?: (selectedItem?: Item<ItemT>) => void;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、{@link DefaultPickerAccessory}がデフォルトで表示されます。
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、{@link SelectPickerItems}がデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * Keyを作成する関数
   */
  keyExtractor?: ((item: Item<ItemT>, index: number) => string) | undefined;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   */
  pickerProps?: Omit<SelectPickerItemsProps<ItemT>, 'items' | 'selectedValue' | 'onValueChange' | 'keyExtractor'>;
  /**
   * PickerBackdropのProps
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

const DEFAULT_DURATION = 300;

export const SelectPicker = <ItemT extends unknown>(props: SelectPickerProps<ItemT>) => {
  const {isVisible, inputValue, handleBackdropPress, onValueChange, open, handleDelete, handleCancel, handleDone} =
    useSelectPicker<ItemT>(props);
  const {
    items,
    selectedItemKey,
    placeholder,
    textInputProps,
    keyExtractor,
    pickerAccessoryComponent,
    pickerItemsComponent,
    textInputComponent,
    pickerItemsContainerProps: {style: pickerItemsContainerStyle, ...pickerItemsContainerProps} = {},
    pickerProps,
    pickerBackdropProps: {
      fadeInDuration = DEFAULT_DURATION,
      fadeOutDuration = DEFAULT_DURATION,
      ...pickerBackdropProps
    } = {},
    pickerContainerProps: {
      style: pickerContainerStyle,
      slideInDuration = DEFAULT_DURATION,
      slideOutDuration = DEFAULT_DURATION,
      ...pickerContainerProps
    } = {},
    pickerAccessoryProps,
  } = props;

  return (
    <>
      <PickerBackdrop
        isVisible={isVisible}
        onPress={handleBackdropPress}
        fadeInDuration={fadeInDuration}
        fadeOutDuration={fadeOutDuration}
        {...pickerBackdropProps}>
        <PickerContainer
          isVisible={isVisible}
          style={[styles.pickerContainer, pickerContainerStyle]}
          slideInDuration={slideInDuration}
          slideOutDuration={slideOutDuration}
          {...pickerContainerProps}>
          {pickerAccessoryComponent ? (
            pickerAccessoryComponent
          ) : (
            <DefaultPickerAccessory
              onDelete={handleDelete}
              onCancel={handleCancel}
              onDone={handleDone}
              {...pickerAccessoryProps}
            />
          )}
          <View {...pickerItemsContainerProps}>
            {pickerItemsComponent ? (
              pickerItemsComponent
            ) : (
              <SelectPickerItems
                selectedValue={selectedItemKey}
                items={items}
                onValueChange={onValueChange}
                keyExtractor={keyExtractor}
                {...pickerProps}
              />
            )}
          </View>
        </PickerContainer>
      </PickerBackdrop>
      <Pressable onPress={open} testID="pressableContainer">
        <View pointerEvents="box-only">
          {textInputComponent ? (
            textInputComponent
          ) : (
            // テキスト入力とスタイルを合わせるために、TextではなくTextInputを使用する
            <TextInput placeholder={placeholder} value={inputValue} editable={false} {...textInputProps} />
          )}
        </View>
      </Pressable>
    </>
  );
};

const styles = StyleSheet.create({
  pickerContainer: {
    backgroundColor: 'white',
  },
});
