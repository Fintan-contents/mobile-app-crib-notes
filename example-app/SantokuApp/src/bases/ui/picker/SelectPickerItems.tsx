import {PickerProps} from '@react-native-picker/picker';
import {log} from 'bases/logging';
import React from 'react';
import {Platform, StyleProp, TextStyle} from 'react-native';

import {Item} from './SelectPicker';

export type SelectPickerItemsProps<ItemT> = Omit<
  PickerProps<React.Key | ItemT>,
  'enabled' | 'mode' | 'prompt' | 'dropdownIconColor' | 'dropdownIconRippleColor' | 'onFocus' | 'onBlur' | 'style'
> & {
  items: Item<ItemT>[];
  keyExtractor?: ((item: Item<ItemT>, index: number) => string) | undefined;
  /**
   * Pickerのスタイル
   * @platform ios
   */
  style?: StyleProp<TextStyle>;
  /**
   * アイテムの高さ
   * @platform android
   */
  itemHeight?: number;
  /**
   * 選択状態のアイテムのテキストカラー
   * @platform android
   */
  activeColor?: string;
  /**
   * 未選択状態のアイテムのテキストカラー
   * @platform android
   */
  inactiveColor?: string;
  /**
   * FlatListのtestID
   * @platform android
   */
  flatListTestID?: string;
  /**
   * SeparatorのtestID
   * @platform android
   */
  separatorTestID?: string;
  /**
   * SelectPickerItemのPressableのtestIDプレフィックス
   * @platform android
   */
  itemPressableTestIDPrefix?: string;
  /**
   * SelectPickerItemのTextのtestIDプレフィックス
   * @platform android
   */
  itemTextTestIDPrefix?: string;
};

export const SelectPickerItems = <ItemT extends unknown>(props: SelectPickerItemsProps<ItemT>) => {
  React.useEffect(() => {
    log.warn(`SelectPickerItems is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
