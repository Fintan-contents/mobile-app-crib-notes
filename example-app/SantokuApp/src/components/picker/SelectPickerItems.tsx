import {PickerProps} from '@react-native-picker/picker';
import {log} from 'framework/logging';
import React from 'react';
import {Platform} from 'react-native';

import {Item} from './SelectPicker';

export type SelectPickerItemsProps = PickerProps<React.Key> & {
  items: Item<React.Key>[];
  keyExtractor?: ((item: React.Key, index: number) => React.Key) | undefined;

  // android
  initialValue?: React.Key;
  itemHeight?: number;
  preferredNumVisibleRows?: number;
};

export const SelectPickerItems = (props: SelectPickerItemsProps) => {
  React.useEffect(() => {
    log.warn(`SelectPickerItems is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
