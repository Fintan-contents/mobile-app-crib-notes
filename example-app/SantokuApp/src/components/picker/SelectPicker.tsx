import React from 'react';
import {Pressable, StyleSheet, TextInput, TextInputProps as RNETextInputProps, View, ViewProps} from 'react-native';

import {DefaultPickerAccessory, DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdrop, PickerBackdropProps} from './PickerBackdrop';
import {PickerContainer, PickerContainerProps} from './PickerContainer';
import {SelectPickerItems, SelectPickerItemsProps} from './SelectPickerItems';
import {useSelectPickerUseCase} from './useSelectPickerUseCase';

export type Item<T> = {
  label: string;
  value: T;
  inputLabel?: string;
  key?: React.Key;
  color?: string;
  fontFamily?: string;
};

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;
export type SelectPickerProps<ItemT> = {
  /**
   * SelectPickerに表示するアイテム
   */
  items: Item<ItemT>[];
  /**
   * 選択されたアイテムのKey
   */
  selectedItemKey?: React.Key | ItemT;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (selectedItem?: Item<ItemT>) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   */
  onDismiss?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDelete?: (selectedItem?: Item<ItemT>) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onCancel?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDone?: (selectedItem?: Item<ItemT>) => void;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、{@link DefaultPickerAccessory}がデフォルトで表示されます。
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、{@link SelectPickerItems}がデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * Keyを作成する関数
   */
  keyExtractor?: ((item: Item<ItemT>, index: number) => string) | undefined;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   */
  pickerProps?: Omit<SelectPickerItemsProps<ItemT>, 'items' | 'selectedValue' | 'onValueChange' | 'keyExtractor'>;
  /**
   * PickerBackdropのProps
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

export const SelectPicker = <ItemT extends unknown>(props: SelectPickerProps<ItemT>) => {
  const {
    isVisible,
    inputValue,
    handleBackdropPress,
    pickerBackdropEntering,
    pickerBackdropExiting,
    pickerContainerEntering,
    pickerContainerExiting,
    onValueChange,
    open,
    handleDelete,
    handleCancel,
    handleDone,
  } = useSelectPickerUseCase<ItemT>(props);
  const {
    items,
    selectedItemKey,
    placeholder,
    textInputProps,
    keyExtractor,
    pickerAccessoryComponent,
    pickerItemsComponent,
    textInputComponent,
    pickerItemsContainerProps: {style: pickerItemsContainerStyle, ...pickerItemsContainerProps} = {},
    pickerProps,
    pickerBackdropProps: {entering: backdropEntering, exiting: backdropExiting, ...pickerBackdropProps} = {},
    pickerContainerProps: {
      entering: containerEntering,
      exiting: containerExiting,
      style: pickerContainerStyle,
      ...pickerContainerProps
    } = {},
    pickerAccessoryProps,
  } = props;

  return (
    <>
      <PickerBackdrop
        isVisible={isVisible}
        onPress={handleBackdropPress}
        entering={pickerBackdropEntering}
        exiting={pickerBackdropExiting}
        {...pickerBackdropProps}>
        <PickerContainer
          isVisible={isVisible}
          entering={pickerContainerEntering}
          exiting={pickerContainerExiting}
          style={StyleSheet.flatten([styles.pickerContainer, pickerContainerStyle])}
          {...pickerContainerProps}>
          {pickerAccessoryComponent ? (
            pickerAccessoryComponent
          ) : (
            <DefaultPickerAccessory
              onDelete={handleDelete}
              onCancel={handleCancel}
              onDone={handleDone}
              {...pickerAccessoryProps}
            />
          )}
          <View {...pickerItemsContainerProps}>
            {pickerItemsComponent ? (
              pickerItemsComponent
            ) : (
              <SelectPickerItems
                selectedValue={selectedItemKey}
                items={items}
                onValueChange={onValueChange}
                keyExtractor={keyExtractor}
                {...pickerProps}
              />
            )}
          </View>
        </PickerContainer>
      </PickerBackdrop>
      <Pressable onPress={open} testID="pressableContainer">
        <View pointerEvents="box-only">
          {textInputComponent ? (
            textInputComponent
          ) : (
            // テキスト入力とスタイルを合わせるために、TextではなくTextInputを使用する
            <TextInput placeholder={placeholder} value={inputValue} editable={false} {...textInputProps} />
          )}
        </View>
      </Pressable>
    </>
  );
};

const styles = StyleSheet.create({
  pickerContainer: {
    backgroundColor: 'white',
  },
});
