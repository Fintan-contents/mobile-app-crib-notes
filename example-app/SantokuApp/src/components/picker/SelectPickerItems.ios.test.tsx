import {fireEvent, render, screen} from '@testing-library/react-native';
import React from 'react';

import {Item} from './SelectPicker';
import {SelectPickerItemsProps} from './SelectPickerItems';
import {SelectPickerItems} from './SelectPickerItems.ios';

describe('SelectPickerItems only with required props', () => {
  it('renders successfully with more than one item', () => {
    const items = [
      {value: '1', label: 'test1'},
      {value: '2', label: 'test2'},
    ];
    render(<SelectPickerItems selectedValue="1" items={items} testID="SelectPickerItems" />);
    const selectPickerItems = screen.queryByTestId('SelectPickerItems');
    expect(screen).toMatchSnapshot('SelectPickerItems with more than one item.');
    expect(selectPickerItems).not.toBeNull();
  });

  it('renders successfully if item does not exit', () => {
    render(<SelectPickerItems selectedValue="1" items={[]} testID="SelectPickerItems" />);
    const selectPickerItems = screen.queryByTestId('SelectPickerItems');
    expect(screen).toMatchSnapshot('SelectPickerItems without item.');
    expect(selectPickerItems).not.toBeNull();
  });
});

describe('SelectPickerItems with all props', () => {
  it('should be applied properly', () => {
    const items = [
      {value: '1', label: 'test1', key: 'key1', color: 'red', fontFamily: 'Roboto'},
      {value: '2', label: 'test2', color: 'yellow', fontFamily: 'SFProText'},
    ];
    const onValueChange = jest.fn();
    const keyExtractor = jest.fn((item: Item<string>, index: number) => `${item.value}-${index}`);
    /**
     * 以下のPropは自動テストで検証できなかったため除外
     * - style
     * - itemStyle
     * - accessibilityLabel
     */
    render(
      <SelectPickerItems
        selectedValue="2"
        items={items}
        onValueChange={onValueChange}
        numberOfLines={5}
        keyExtractor={keyExtractor}
        testID="SelectPickerItems"
      />,
    );
    expect(screen).toMatchSnapshot('SelectPickerItems with all props.');
    const selectPickerItems = screen.getByTestId('SelectPickerItems');

    fireEvent(selectPickerItems, 'onValueChange');

    const selectPickerItemsProps = selectPickerItems.props as SelectPickerItemsProps<string>;
    const selectedIndex = selectPickerItems.props.selectedIndex as number;
    expect(selectedIndex).toBe(1);
    // itemsは、Itemの各Propを検証できなかったため、件数のみ検証する
    expect(selectPickerItemsProps.items.length).toBe(2);
    expect(onValueChange).toHaveBeenCalled();
    expect(selectPickerItemsProps.numberOfLines).toBe(5);
    expect(keyExtractor).toHaveBeenNthCalledWith(
      1,
      {value: '1', label: 'test1', key: 'key1', color: 'red', fontFamily: 'Roboto'},
      0,
    );
    expect(keyExtractor).toHaveBeenNthCalledWith(
      2,
      {value: '2', label: 'test2', color: 'yellow', fontFamily: 'SFProText'},
      1,
    );
  });
});
