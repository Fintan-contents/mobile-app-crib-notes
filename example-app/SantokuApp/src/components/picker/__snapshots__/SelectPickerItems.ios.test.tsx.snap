// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectPickerItems only with required props renders successfully if item does not exit: SelectPickerItems without item. 1`] = `
<View>
  <RNCPicker
    items={Array []}
    numberOfLines={1}
    onChange={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
    testID="SelectPickerItems"
  />
</View>
`;

exports[`SelectPickerItems only with required props renders successfully with more than one item: SelectPickerItems with more than one item. 1`] = `
<View>
  <RNCPicker
    items={
      Array [
        Object {
          "label": "test1",
          "testID": undefined,
          "textColor": undefined,
          "value": "1",
        },
        Object {
          "label": "test2",
          "testID": undefined,
          "textColor": undefined,
          "value": "2",
        },
      ]
    }
    numberOfLines={1}
    onChange={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
    testID="SelectPickerItems"
  />
</View>
`;

exports[`SelectPickerItems with all props should be applied properly: SelectPickerItems with all props. 1`] = `
<View>
  <RNCPicker
    items={
      Array [
        Object {
          "label": "test1",
          "testID": undefined,
          "textColor": 4294901760,
          "value": "key1",
        },
        Object {
          "label": "test2",
          "testID": undefined,
          "textColor": 4294967040,
          "value": "2",
        },
      ]
    }
    numberOfLines={5}
    onChange={[Function]}
    selectedIndex={1}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
    testID="SelectPickerItems"
  />
</View>
`;
